ansible_collections/arista/avd/plugins/module_utils/eos_designs_facts.py

```diff
@@ -0,0 +1,1306 @@
+
+            if len(lt_groups) > 0:
+                for lt_group in lt_groups:
+                    lt_group['recovery_delay'] = lt_group.get('recovery_delay', default_recovery_delay)
+                    link_tracking_groups.append(lt_group)
+            else:
+                link_tracking_groups.append({"name": "LT_GROUP1", "recovery_delay": default_recovery_delay})
+
+            return link_tracking_groups
+
+        return None
+
+    @cached_property
+    def lacp_port_id(self):
+        if get(self._switch_data_combined, "lacp_port_id_range.enabled") is True:
+            node_group_length = max(len(self._switch_data_node_group_nodes), 1)
+            lacp_port_id = {}
+            switch_id = self.id
+            port_range = int(get(self._switch_data_combined, 'lacp_port_id_range.size', default=128))
+            port_offset = int(get(self._switch_data_combined, 'lacp_port_id_range.offset', default=0))
+            lacp_port_id['begin'] = 1 + (((switch_id - 1) % node_group_length) * port_range) + port_offset
+            lacp_port_id['end'] = (((switch_id - 1) % node_group_length + 1) * port_range) + port_offset
+            return lacp_port_id
+
+        return None
+
+    @cached_property
+    def _any_network_services(self):
+        '''
+        Returns True if either L1, L2 or L3 network_services are enabled
+        '''
+        return (
+            self.network_services_l1 is True
+            or self.network_services_l2 is True
+            or self.network_services_l3 is True
+        )
+
+    @cached_property
+    def filter_tenants(self):
+        if self._any_network_services:
+            return get(self._switch_data_combined, "filter.tenants", default=["all"])
+        return None
+
+    @cached_property
+    def always_include_vrfs_in_tenants(self):
+        if self._any_network_services:
+            return get(self._switch_data_combined, "filter.always_include_vrfs_in_tenants")
+        return None
+
+    @cached_property
+    def filter_tags(self):
+        if self._any_network_services:
+            return get(self._switch_data_combined, "filter.tags", default=["all"])
+        return None
+
+    @cached_property
+    def virtual_router_mac_address(self):
+        if self._any_network_services:
+            return get(self._switch_data_combined, "virtual_router_mac_address")
+        return None
+
+    @cached_property
+    def _vlans(self):
+        if self._any_network_services:
+            vlans = []
+            match_tags = self.filter_tags
+            if self.group is not None:
+                match_tags.append(self.group)
+
+            network_services_keys = get(self._hostvars, "network_services_keys", default=[])
+            for network_services_key in self._natural_sort(network_services_keys, "name"):
+                network_services_key_name = network_services_key.get("name")
+                if network_services_key_name is not None and get(self._hostvars, network_services_key_name) is not None:
```

Comment:
for this kind of methods with a lots of nested block, a static code analysis tool would generally end up complaining about too many nested if/for/..

a solution here to save on indentation in the for loop could be something like
```
for ...
  if network_service_key_name is None or get(self._hostvars, network_services_key_name) is None:
    continue # move to next element of the list
  tenants = self._convert_dicts(tenants, 'name')
```

sometimes it make things more readable rather than too many nested for/if loop

Reply:
I have tried to fix the worst ones